python/lsst/sconsUtils/builders.py

```diff
@@ -237,7 +237,7 @@ def ProductDir(env, product):
     global _productDirs
     try:
         _productDirs
-    except:
+    except Exception:
```

Comment:
This change is fine but I'm puzzled what the `try` is for. When the module is first loaded _productDirs is sure to be undefined so the `try` is sure to run, attempting to compute `_productDirs`. Naively I would expect the outer try/except block to do nothing.

Reply:
First time through it's not set.

Reply:
First time through the try/except fires. How is that different than getting rid of the outer try/except and just setting `_productDirs`?

Reply:
This logic is 6 years old. We could work around it by setting the variable to none at module scope and then checking for `is None` rather than doing the `try` block. I'm inclined to leave it alone for the moment.

Reply:
Agreed, I guess. I am just trying to understand if the try/except is actually doing anything at all. Have you tried removing it to see if it still works?